vmm_tests/vmm_tests/tests/tests/x86_64/openhcl_linux_direct.rs

```diff
@@ -396,6 +396,90 @@ async fn openhcl_linux_storvsp_dvd(config: PetriVmConfigOpenVmm) -> Result<(), a
     Ok(())
 }
 
+/// Test an OpenHCL Linux direct VM with a SCSI DVD assigned to VTL2, using NVMe
+/// backing, and vmbus relay. This should expose a DVD to VTL0 via vmbus.
+#[openvmm_test(openhcl_linux_direct_x64)]
+async fn openhcl_linux_storvsp_dvd_nvme(config: PetriVmConfigOpenVmm) -> Result<(), anyhow::Error> {
+    const NVME_INSTANCE: Guid = guid::guid!("dce4ebad-182f-46c0-8d30-8446c1c62ab3");
+    let vtl2_nsid = 1;
+    let nvme_disk_sectors: u64 = 0x10000;
+    let sector_size = 512;
+
+    let vtl2_lun = 5;
+    let scsi_instance = Guid::new_random();
+
+    let mut vtl2_settings = None;
+
+    let (vm, agent) = config
+        .with_vmbus_redirect()
+        .with_custom_config(|c| {
+            c.vpci_devices.extend([VpciDeviceConfig {
+                vtl: DeviceVtl::Vtl2,
+                instance_id: NVME_INSTANCE,
+                resource: NvmeControllerHandle {
+                    subsystem_id: NVME_INSTANCE,
+                    max_io_queues: 64,
+                    msix_count: 64,
+                    namespaces: vec![NamespaceDefinition {
+                        nsid: vtl2_nsid,
+                        disk: (LayeredDiskHandle::single_layer(RamDiskLayerHandle {
+                            len: Some(nvme_disk_sectors * sector_size),
+                        })
+                        .into_resource()),
+                        read_only: false,
+                    }],
+                }
+                .into_resource(),
+            }]);
+        })
+        .with_custom_vtl2_settings(|v| {
+            v.dynamic.as_mut().unwrap().storage_controllers.push(
+                vtl2_settings_proto::StorageController {
+                    instance_id: scsi_instance.to_string(),
+                    protocol: vtl2_settings_proto::storage_controller::StorageProtocol::Scsi.into(),
```

Comment:
I'd love to see a gen1 vm equivalent of this test as well (as a separate follow up, if we go that route)